pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -126,6 +127,7 @@ func NewScheduler(
 			return np.Name, corev1.ResourceList(np.Spec.Limits)
 		}),
 		clock:                clock,
+		cache:                cache,
```

Comment:
Given that we already have a pod cache -- does it make sense to share this cache across both node requirements and pod requirements so that we keep a single cache and just have different data in it?